### PR TITLE
feat: 530 set up express REST API

### LINKED ISSUE #530: [node-pod] 0. Generate project and establish integration point
# Background

To make tasks parallelizable, we should create kits via integration branches to make PRs smaller and easier to read while not breaking the upstream build. Integration branches may be cleaner and easier to implement in isolation as kits are isolated to their own directory anyway.

# Acceptance

- [ ] Generate new project in the `starters/` creating a name that includes `<framework>-<special-tech>-<store>` format
- [x] Change the name and description in the package.json
- [x] Merge change into an integration branch

### PR DESCRIPTION
**Pull request by TapaiBalazs**
## Type of change

- [ ] Documentation change
- [ ] Bug fix
- [X] Feature

## Summary of change

- Created an express-typescript API base setup
- Jest unit testing support
- Supertest API testing support
- set up linter
- set up prettier

TODO:
- [x] Readme.md documentation
- [x] Additional project tooling (technology) setup

## Checklist

- [X] These changes follow the [contributing guidelines](https://github.com/thisdot/starter.dev/blob/main/CONTRIBUTING.md)
- [x] This starter kit has been approved by the maintainers
- [X] Changes for this new starter kit are being pushed to an integration branch instead of main
- [x] All dependencies are version locked
- [X] This fix resolves #530 #531 #535 
- [X] I have verified the fix works and introduces no further errors

### CODE REVIEW

File: starters/express-typescript-typeorm-postgres/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  env: {
+    node: true,
+    jest: true,
+  },
+  ignorePatterns: ['.eslintrc.js'],
+  rules: {
+    '@typescript-eslint/interface-name-prefix': 'off',
+    '@typescript-eslint/explicit-function-return-type': 'off',
+    '@typescript-eslint/explicit-module-boundary-types': 'off',
+    '@typescript-eslint/no-explicit-any': 'off',
```
Comment (BigAB) [inline starters/express-typescript-typeorm-postgres/.eslintrc.js:23]: You   ...like `any`?

> `any` is the mind-killer.
> `any` is the little-death that brings total obliteration.
>                    - Frank Herbert's Dune-of-TypeScript

I mean, it's fine, but are we sure we want to just allow any anytime we want?

Comment (TapaiBalazs) [reply starters/express-typescript-typeorm-postgres/.eslintrc.js:23]: @BigAB do you want warning or error?

Comment (BigAB) [reply starters/express-typescript-typeorm-postgres/.eslintrc.js:23]: I'm actually pretty okay with leaving this `off`, but if it were just based on my personal preference I would go with a warning I guess.

I am curious how the others feel though.

Comment (TapaiBalazs) [reply starters/express-typescript-typeorm-postgres/.eslintrc.js]: set it to error

**Review by BigAB — APPROVED** [review]
LGTM

**Pull request merged**